examples/step-62/step-62.cc

```diff
@@ -1064,17 +1073,19 @@ namespace step62
           {
             point[dim_idx] = probe_positions[position_idx][dim_idx];
           }
-        bool point_in_locally_owned_cell;
+        bool point_in_locally_owned_cell = false;
         {
-          // First we have to find out if the point is in a locally owned cell.
-          auto mapping = StaticMappingQ1<dim>::mapping;
-          const std::pair<typename DoFHandler<dim>::active_cell_iterator,
-                          Point<dim>>
-            cell_point = GridTools::find_active_cell_around_point(mapping,
-                                                                  dof_handler,
-                                                                  point);
-
-          point_in_locally_owned_cell = cell_point.first->is_locally_owned();
+          auto cell_and_ref_point = GridTools::find_active_cell_around_point(
+            cache, point, cell_hint, marked_vertices, tolerance);
+          if (cell_and_ref_point.first != triangulation.end())
+            {
+              if (cell_and_ref_point.first.state() == IteratorState::valid)
```

Comment:
I don't think you need this second check. The first one is already what you need, the second just is a different way of asking the same question.

Reply:
It looks like I need both `ifs`. If I remove the second `if`, I get the following error:
```
--------------------------------------------------------
An error occurred in line <1014> of file </home/daniel/repositories/dealii/include/deal.II/grid/tria_iterator.h> in function                                                                                        
    Accessor &dealii::TriaRawIterator<dealii::CellAccessor<2, 2> >::operator*() [Accessor = dealii::CellAccessor<2, 2>]                                                                                             
The violated condition was:
    Accessor::structure_dimension != Accessor::dimension || state() == IteratorState::valid
Additional information:
    You tried to dereference a cell iterator for which this is not possible. More information on this iterator: level=-2, index=-2, state=invalid                                                                   

Stacktrace:
-----------
#0  step-62.debug: dealii::TriaRawIterator<dealii::CellAccessor<2, 2> >::operator*()
#1  step-62.debug: dealii::TriaRawIterator<dealii::CellAccessor<2, 2> >::operator->()
#2  step-62.debug: step62::ElasticWave<2>::store_frequency_step_data(unsigned int)
#3  step-62.debug: step62::ElasticWave<2>::frequency_sweep()
#4  step-62.debug: step62::ElasticWave<2>::run()
#5  step-62.debug: main
--------------------------------------------------------

Calling MPI_Abort now.
```

Reply:
Apparently, the default constructor of `active_cell_iterator` is not equal to `end()`. I think this is a bug and should be fixed.

Reply:
@luca-heltai I updated the PR and now I use only the second `if`. Maybe we can merge this PR like this and when `active_cell_iterator` is fixed I can change this to use only the first `if`

That's right the default constructor of `active_cell_iterator` is not equal to `end()`. Should I do a separate PR to address this issue?

Reply:
> That's right the default constructor of active_cell_iterator is not equal to end(). Should I do a separate PR to address this issue?

this would be great, and the correct solution.

Reply:
Sounds good! I'll prepare a separate PR